consensus/src/chained_bft/event_processor.rs

```diff
@@ -732,64 +729,25 @@ impl<T: Payload> EventProcessor<T> {
     }
 
     /// Upon (potentially) new commit:
-    /// 0. Verify that this commit is newer than the current root.
-    /// 1. Notify state computer with the finality proof.
+    /// 1. Commit the blocks via block store.
     /// 2. After the state is finalized, update the txn manager with the status of the committed
     /// transactions.
-    /// 3. Prune the tree.
     async fn process_commit(
         &self,
         block_id_to_commit: HashValue,
         finality_proof: LedgerInfoWithSignatures,
     ) {
-        let block_to_commit = match self.block_store.get_block(block_id_to_commit) {
-            Some(block) => block,
-            None => {
-                return;
-            }
-        };
-
-        // First make sure that this commit is new.
-        if block_to_commit.round() <= self.block_store.root().round() {
-            return;
-        }
-
-        // Verify that the ledger info is indeed for the block we're planning to
-        // commit.
-        assert_eq!(
-            finality_proof.ledger_info().consensus_block_id(),
-            block_to_commit.id()
-        );
-
-        let blocks_to_commit = self
+        let blocks_to_commit = match self
             .block_store
-            .path_from_root(block_id_to_commit)
-            .unwrap_or_else(Vec::new);
-
-        let payload_and_output_list = blocks_to_commit
-            .iter()
-            .map(|b| {
-                (
-                    b.payload().unwrap_or(&T::default()).clone(),
-                    Arc::clone(b.output()),
-                )
-            })
-            .collect();
-        if let Err(e) = self
-            .state_computer
-            .commit(payload_and_output_list, finality_proof)
+            .commit(block_id_to_commit, finality_proof)
             .await
         {
-            // We assume that state computer cannot enter an inconsistent state that might
-            // violate safety of the protocol. Specifically, an executor service is going to panic
-            // if it fails to persist the commit requests, which would crash the whole process
-            // including consensus.
-            error!(
-                "Failed to persist commit, mempool will not be notified: {:?}",
```

Comment:
If the commit fails, why shouldn't we panic!?

Reply:
see the comment we had above, it's assumed the whole process would crash

Reply:
Why not also panic here then?  If somehow the other behavior changed, we would want to panic here right?